chugbot/attendance/index.php

```diff
@@ -0,0 +1,238 @@
+<?php
+include_once '../dbConn.php';
+include_once '../functions.php';
+session_start();
+checkLogout();
+
+// if not admin, redirect to appropriate home page
+if(!adminLoggedIn()) {
+    if(roshLoggedIn()) {
+        $roshUrl = urlIfy("../attendance/roshHome.php");
+        header("Location: $roshUrl");
+        exit();
+    }
+    else if(chugLeaderLoggedIn()) {
+        $chugLeaderUrl = urlIfy("../attendance/chugLeaderHome.php");
+        header("Location: $chugLeaderUrl");
+        exit();
+    }
+    else if(camperLoggedIn()) {
+        $camperUrl = urlIfy("../camperHome.php");
+        header("Location: $camperUrl");
+        exit();
+    }
+    else {
+        $baseUrl = baseUrl();
+        header("Location: $baseUrl");
+        exit();
+    }
+}
+
+
+$dbErr = "";
+$sessionId2Name = array();
+$blockId2Name = array();
+$groupId2Name = array();
+$edahId2Name = array();
+$chugId2Name = array();
+$bunkId2Name = array();
+$group2ActiveBlock = array();
+
+fillId2Name(null, $chugId2Name, $dbErr, "chug_id", "chugim", "group_id", "chug_groups");
+fillId2Name(null, $sessionId2Name, $dbErr, "session_id", "sessions");
+fillId2Name(null, $blockId2Name, $dbErr, "block_id", "blocks");
+fillId2Name(null, $groupId2Name, $dbErr, "group_id", "chug_groups");
+fillId2Name(null, $edahId2Name, $dbErr, "edah_id", "edot");
+fillId2Name(null, $bunkId2Name, $dbErr, "bunk_id", "bunks");
+
+
+setup_camp_specific_terminology_constants();
+
+echo headerText("Attendance Admin");
+
+// update saved attendance
+if ($_SERVER["REQUEST_METHOD"] == "POST") {
+    // determine what is being updated/set
+    if(test_post_input("form") == "active-blocks") {
+        $localErr = "";
+        $dbc = new DbConn();
+        // list of applicable chug group ids
+        $groupIds = array_keys($groupId2Name);
+
+        // build SQL
+        $anySet = FALSE;
+        $sql = "UPDATE chug_groups SET active_block_id = ( CASE ";
+        foreach($groupIds as $id) {
+            $block = test_post_input($id);
+            if (!empty($block)) {
+                $sql .= "WHEN (group_id = $id) THEN $block ";
+                $anySet = TRUE;
+            }
+        }
+        $sql .= "ELSE (NULL) END )";
+        // if all are set to null, override built SQL and just clear column
+        if(!$anySet) {
+            $sql = "UPDATE chug_groups SET active_block_id = NULL";
+
```

Comment:
nit: There are a few places with empty lines. If you apply an auto-format, it might clear these up. Definitely not a blocker.